frontend/public/components/cluster-settings/related-objects.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useSelector } from 'react-redux';
+import * as classNames from 'classnames';
+import * as _ from 'lodash-es';
+import { sortable } from '@patternfly/react-table';
+import { Table, TableRow, TableData, RowFunctionArgs } from '../factory';
+import { referenceForModel, K8sResourceKind, K8sKind } from '../../module/k8s';
+import { ResourceLink } from '../utils';
+
+const referenceFromGroupKind = (group: string = '', kind: string = '') => `${group}~${kind}`;
+
+const tableColumnClasses = [
+  classNames('col-lg-5', 'col-md-5', 'col-sm-4', 'col-xs-12'),
+  classNames('col-lg-2', 'col-md-2', 'col-sm-4', 'hidden-xs'),
+  classNames('col-lg-5', 'col-md-5', 'col-sm-4', 'hidden-xs'),
+];
+
+const Header = () => [
+  {
+    title: 'Name',
+    sortField: 'name',
+    transforms: [sortable],
+    props: { className: tableColumnClasses[0] },
+  },
+  {
+    title: 'Kind',
+    sortField: 'resource',
+    transforms: [sortable],
+    props: { className: tableColumnClasses[1] },
+  },
+  {
+    title: 'Namespace',
+    sortField: 'namespace',
+    transforms: [sortable],
+    props: { className: tableColumnClasses[2] },
+  },
+];
+
+const Row: React.FC<RowFunctionArgs> = ({ obj, index, key, style, customData }) => {
+  const { name, resource, namespace, group } = obj;
+  const uid = `${name}-${resource}-${namespace}-${group}`;
+  const model = customData[referenceFromGroupKind(group, resource)];
+
+  const gsv = model ? referenceForModel(model) : '';
+  return (
+    <TableRow id={uid} index={index} trKey={key} style={style}>
+      <TableData className={tableColumnClasses[0]}>
+        <ResourceLink kind={gsv} name={name} namespace={namespace} />
+      </TableData>
+      <TableData className={tableColumnClasses[1]}>{resource}</TableData>
+      <TableData className={tableColumnClasses[2]}>
+        {!_.isEmpty(namespace) ? <ResourceLink kind="Namespace" name={namespace} /> : ''}
+      </TableData>
+    </TableRow>
+  );
+};
+
+const RelatedObjects: React.FC<RelatedObjectsProps> = (props) => {
+  // All the models registered in the cluster
+  const { data } = props;
+  const models: K8sKind[] = _.values(
+    useSelector(({ k8s }) => k8s.getIn(['RESOURCES', 'models'])).toJS(),
+  );
+  const uniqueKinds = _.uniqBy(data, (item) => item.resource);
+  const relatedModels = models.reduce((acc, currModel) => {
+    // Check if current model is related to any of the data
+    // If so add a entry to acc that maps from group.resource to the resource
+    const relatedData = _.find(
+      uniqueKinds,
+      (item) =>
+        (currModel.kind === item.resource || currModel.plural === item.resource) &&
+        (currModel?.apiGroup ? currModel?.apiGroup === item?.group : true),
+    );
```

Comment:
This logic won't find the right version.

We should

1. Use the group's `preferredVersion` if the kind exists in that version.

https://github.com/openshift/console/blob/ed152e009843fe6e2520a315d7836e146e12aec1/frontend/public/module/k8s/get-resources.ts#L101

2. Fall back to the sorting from #5355 to find the most recent stable version.

Reply:
The preferred version in the store right now does not store sufficient information. I have introduced a groupVersionMap to k8s  which has the preferred version as well as the other version for that particular group. 
1) First, try with the preferred version.
2) If not found then sort the other versions available and try out with each of them.
3) If still not found it is a K8s native kind so use version as  v1 and has no group and match kind with plural or kind.